backend/rates/history.go

```diff
@@ -167,35 +167,35 @@ func (updater *RateUpdater) backfillHistory(ctx context.Context, coin, fiat stri
 		// When to update next, after this loop iteration is done.
 		untilNext := time.Duration(1+rand.Intn(5)) * time.Second
 
+		// We want hourly rates for the last 90 days and daily past that.
+		// 90 days is the max interval CoinGecko responds with hourly timeseries to.
 		end := updater.HistoryEarliestTimestamp(coin, fiat)
 		var start time.Time
-		if end.IsZero() {
-			// First time; don't have historical data yet.
+		switch {
+		// First time; don't have historical data yet.
+		case end.IsZero():
 			end = time.Now()
-			// 90 days is the max interval which CoinGecko responds to with hourly timeseries.
-			// Make sure we get hourly rates by requesting 90 days minus 1 hour, just in case.
-			start = end.Add(-90*24*time.Hour + time.Hour)
-		} else {
-			// End at the day earlier than the last known timestamp.
-			// We want daily rates for timestamps past the first 90 days: use annual intervals.
+			start = end.Add(-90*24*time.Hour + time.Hour) // 1 hour diff just in case
+		// Less than 90 days: still want hourly rates.
+		case time.Since(end) < 90*24*time.Hour:
+			till90days := 90*24*time.Hour - time.Since(end)
```

Comment:
isn't `start = end.Add(-90*24*time.Hour)` easier? :D

Reply:
:+1: although + extra 1h like in the zero case.

Reply:
actually, no need for +1h. it's always less than 90days here.